src/parselmouth/PointProcess.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2019-2021  Yannick Jadoul
+ *
+ * This file is part of Parselmouth.
+ *
+ * Parselmouth is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Parselmouth is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Parselmouth.  If not, see <http://www.gnu.org/licenses/>
+ */
+
+#include "Parselmouth.h"
+
+#include "TimeClassAspects.h"
+
+#include <praat/fon/PointProcess.h>
+#include <praat/fon/Pitch_to_PointProcess.h>
+#include <praat/fon/VoiceAnalysis.h>
+
+#include <vector>
+#include <tuple>
+
+#include <pybind11/stl.h>
+
+namespace py = pybind11;
+using namespace py::literals;
+using namespace std::string_literals;
+
+namespace parselmouth {
+
+PRAAT_CLASS_BINDING(PointProcess) {
+	addTimeFunctionMixin(*this);
+
+	// FORM(NEW1_PointProcess_createEmpty, U"Create an empty PointProcess", U"Create empty PointProcess...")
+	def(py::init([](double startTime, double endTime, integer initialMaxnt) {
+			if (endTime <= startTime)
+				Melder_throw(U"The end time should be greater than the start time.");
+			return PointProcess_create(startTime, endTime, initialMaxnt);
+		}),
+		"start_time"_a = 0.0, "end_time"_a = 1.0, "initialMaxnt"_a = 0);
+
+	//FORM(NEW1_PointProcess_createPoissonProcess, U"Create Poisson process", U"Create Poisson process...")
+	// WORD(name, U"Name", U"poisson")
+	// REAL(startTime, U"Start time (s)", U"0.0")
+	// REAL(endTime, U"End time (s)", U"1.0")
+	// POSITIVE(density, U"Density (/s)", U"100.0")
+
+	def_static("from_pitch",
```

Comment:
Agree on `Pitch` having `to_point_process`. Thought about it but didn't pull the trigger 'cuz it's not "my" file ;-) I'll add that (and similar others if any)

Reply:
Oh, no worries about that!

It's kind of the nature of how all these classes work together. Decoupling them would be nice, but I'm not entirely sure how to do that (except for Praat's solution of just having everything as a freestanding function, rather than methods; but I don't like that API design too much either).